src/variable.h

```diff
@@ -350,6 +352,74 @@ Variable makeVariable(Tag tag, const Dimensions &dimensions, Args &&... args) {
   }
 }
 
+
+inline void incCoord(const std::vector<size_t> limits, std::vector<size_t>& coords) {
```

Comment:
- For indices, this library is using `gsl::index` everywhere, instead of `size_t`.
- This should not be inline, I think?

Reply:
It is inline because it is in the header, and I want this small function to be close to its usage. Should I wrap it into the lambda better?

Reply:
- Put it in a `namespace detail {}`.
- Move the implementation into the source?